mixins/loading-complete/loading-complete-mixin.js

```diff
@@ -1,21 +1,32 @@
-export const LoadingCompleteMixin = (superclass) => class extends superclass {
+import { dedupeMixin } from '@open-wc/dedupe-mixin';
 
-	constructor() {
-		super();
-		this._loadingCompletePromise = new Promise(resolve => {
-			this._loadingCompleteResolve = resolve;
-		});
+export const LoadingCompleteMixin = dedupeMixin((superclass) => class extends superclass {
+
+	#loadingCompleteResolve;
+
+	// eslint-disable-next-line sort-class-members/sort-class-members
+	#loadingCompletePromise = Object.hasOwn(this.constructor.prototype, 'getLoadingComplete')
```

Comment:
@dbatiste This is safe now, right?! 😆

Reply:
In BSI context, yeah. In a FRA context, it would require their build process to include the Babel plugin for transpiling. So if this would ever make its way into a FRA, that would be a risk.

Reply:
Yeah, I would probably avoid `hasOwn` if there is any worry it will be pulled into other things outside of BSI.